xclim/indices/generic.py

```diff
@@ -453,9 +457,17 @@ def spell_mask(
         if not np.isscalar(thresh):
             mask = getattr(mask, var_reducer)("variable")
         # True for all days part of a spell that respected the condition (shift because of the two rollings)
-        is_in_spell = (mask.rolling(time=window).sum() >= 1).shift(time=-(window - 1))
+        is_in_spell = (mask.rolling(time=window).sum() >= 1).shift(
+            time=-(window - 1), fill_value=False
+        )
         # Cut back to the original size
         is_in_spell = is_in_spell.isel(time=slice(0, data.time.size))
+
+    if min_gap > 1:
+        is_in_spell = rl.runs_with_holes(is_in_spell, 1, ~is_in_spell, min_gap).astype(
+            bool
+        )
```

Comment:
I think could be a if/else logic for `min_gap` which would be more efficient than this.
```suggestion
    if min_gap ==1 :
        # rest of the code
    else :
        mask = compare(data, op, thresh)
        is_in_spell = rl.runs_with_holes(mask, window, ~mask, min_gap).astype(
            bool
        )
```

This would be more efficient than first computing `is_in_spell`, just use directly `mask` instead. Now `min_gap` works on the condition instead of the spells, but I think it's equivalent?

Reply:
I'm not sure it is equivalent, or maybe I'm not understanding the suggestion.

I think you are right for the window == 1 case, and the code already does what you suggest for that case. However when the spell has criteria on a moving-window statistic, we need to put True on all days within each spell before measuring the gaps, no ?

In the `win_reducer='mean'` , `window=3` case for example:
```
mask         = [F F F T F F F T F]
is_in_spell = [F F T T T F T T T]
```

Reply:
right I always forget about more complicated spells. I guess my point would be OK for the fast-track logic, but we can leave this as it is